blockdag/blockwindow.go

```diff
@@ -53,13 +54,37 @@ func (window blockWindow) minMaxTimestamps() (min, max int64) {
 	return
 }
 
+var blockWindowBigIntPool = sync.Pool{
+	New: func() interface{} {
+		return big.NewInt(0)
+	},
+}
+
+func acquireBigInt() *big.Int {
+	return blockWindowBigIntPool.Get().(*big.Int)
+}
+
+func releaseBigInt(toRelease *big.Int) {
+	toRelease.SetInt64(0)
+	blockWindowBigIntPool.Put(toRelease)
+}
+
 func (window blockWindow) averageTarget() *big.Int {
 	averageTarget := big.NewInt(0)
+
+	target := acquireBigInt()
 	for _, node := range window {
-		target := util.CompactToBig(node.bits)
+		util.CompactToBigWithDestination(node.bits, target)
 		averageTarget.Add(averageTarget, target)
 	}
-	return averageTarget.Div(averageTarget, big.NewInt(int64(len(window))))
+
+	// Reuse `target` to avoid a big.Int allocation
+	windowLen := target
```

Comment:
Better utilize the pool rather then re-use directly.
Also, please defer the release of all BigInts